[Program Card Images] Have image upload be the second step in the program creation flow

caitlinshk opened the pull request:
### Description

We want to encourage admins to upload images for all their programs. To help with this, this PR adds image upload as a second step in the program creation flow.

Previously, once an admin creates a new program and fills in the details, they'd be taken to the main program edit page. With this PR, once the admin creates a new program, they'll be taken to the image upload page instead.

When an admin chooses "Edit program details" on an existing draft program, that page will *also* redirect to the image upload page once the details have been saved.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Instructions for manual testing

1. Enable the PROGRAM_CARD_IMAGES flag.
2. On the program dashboard, click "Create new program".
3. Fill in some program details.
4. Verify the button on the bottom of the page says "Save and continue to next step".
5. Click that button.
6. Verify that you're taken to the image upload page.


### Issue(s) this completes

Epic #5676 

Fixes #6000

caitlinshk commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/civiform/civiform/6366?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #6341** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6341?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
    * **PR #6314** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6314?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
      * **PR #6325** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6325?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
        * **PR #6365** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6365?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
          * **PR #6366** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6366?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/6366?utm_source=stack-comment).

rockycodes commented on browser-test/src/support/admin_programs.ts:
```diff
@@ -98,8 +98,13 @@ export class AdminPrograms {
-   * page is going to be block edit page.
+   * Creates program with given name.
+   *
+   * @param {boolean} submitNewProgram - If true, the new program will be submitted
```
praise: clever refactor!

rockycodes reviewed (commented):
This looks great, but I have some questions about the overall flow before approving.

1) Going through the flow now I find myself a little lost after uploading the image. Is it in the designs to add a button at the bottom of the image page to keep going to the program edit view? If it is, I think it should be included in this PR. If it's not, I'm ok to approve this for now, but we should go back to design and ask about that.
2) I tested out the "edit program details" as well and it skips the image page and goes directly to editing the program. Is that what we want? Should the button at the bottom still say "Save and continue to next step"?

caitlinshk commented:
> 1. Going through the flow now I find myself a little lost after uploading the image. Is it in the designs to add a button at the bottom of the image page to keep going to the program edit view? If it is, I think it should be included in this PR. If it's not, I'm ok to approve this for now, but we should go back to design and ask about that.

Definitely agree with you. I talked with Ron and Shreya, and we decided to remove the breadcrumbs in the top-left of the page and replace them with a Back button. The breadcrumbs only make sense if all of the admin pages use them, so it's better to not use them at all and wait until we can update all admin pages at once.

I'm going to write a separate PR to replace breadcrumbs with a back button, I added a TODO in this PR.

> 2. I tested out the "edit program details" as well and it skips the image page and goes directly to editing the program. Is that what we want? Should the button at the bottom still say "Save and continue to next step"?

Whoops good catch! I updated the "Edit program details" page to redirect to the image upload page if the images flag is on. (And confirmed with Ron & Shreya that this page *should* also redirect to the image upload page.)
